src/main/java/com/sap/cloudfoundry/client/facade/adapters/ApplicationLogEntity.java

```diff
@@ -9,29 +9,34 @@
 @Value.Immutable
 @Value.Enclosing
 @JsonDeserialize(as = ImmutableApplicationLogEntity.class)
-public interface ApplicationLogEntity {
+public abstract class ApplicationLogEntity implements Comparable<ApplicationLogEntity> {
 
-    String getTimestamp();
+    public abstract Long getTimestamp(); //in nanoseconds
```

Comment:
Remove the comment and maybe refactor the method name `getTimestampInNanoseconds()`